File: java/src/test/java/problems1_100/PalindromeNumberTest.java
package problems1_100;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class PalindromeNumberTest extends SampledTest {
    @Test
    public void test() {
        for (PalindromeNumber solution : PalindromeNumber.values()) {
            for (PalindromeNumberSample sample : getSamples(PalindromeNumberSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.isPalindrome(sample.input.x))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/PalindromeNumberTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class PalindromeNumberTest : SampledTest() {
    @Test
    fun test() =
        PalindromeNumber.entries.forEach { solution ->
            getSamples(Array<PalindromeNumberSample>::class.java)
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.isPalindrome(input.x))
                        .isEqualTo(output)
                }
        }
}
